apps/src/feedbackBlocks.js

```diff
@@ -51,71 +52,65 @@ var FeedbackBlocks = function(options, missingRequiredBlocks, missingRecommended
-  this.iframe.setAttribute('allowtransparency', 'true');
+  this.div.setAttribute('id', 'feedbackBlocks');
 
-  this.div.appendChild(this.iframe);
+  // Will be set by this.render()
+  this.blockSpaceEditor = undefined;
 };
 
 module.exports = FeedbackBlocks;
 
-/**
- * Generate a url (with params) for the iFrame that's going to hold the
- * blocks.
- * TODO(elijah) replace this whole thing with an implementation that
- * doesn't require an iframe, since we can now have multiple blocklys on
- * the same page
- */
-FeedbackBlocks.prototype.readonlyTemplateUrl_ = function () {
-  var params = {
-    app: this.iframeOptions.app,
-    js_locale: this.iframeOptions.options.locale,
-    locale_dir: this.iframeOptions.options.localeDirection
-  };
-
-  // Simple polyfill for $.params
-  var paramString = Object.keys(params).map(function (key) {
-    return key + "=" + encodeURIComponent(params[key]);
-  }).join("&");
-
-  return "/readonly_template?" + paramString;
-};
+FeedbackBlocks.prototype.render = function () {
+  // Only render if this.div exists in the DOM
+  var div = document.getElementById("feedbackBlocks");
+  if (div !== this.div) {
+    return;
+  }
 
-FeedbackBlocks.prototype.show = function() {
-  var iframeOptions = this.iframeOptions;
-  var iframe = this.iframe;
-  iframe.setAttribute('src', this.readonlyTemplateUrl_());
-  iframe.onload = function () {
-    this[iframeOptions.app + "Main"](iframeOptions.options);
-  }.bind(iframe.contentWindow);
+  // This is sort of a nasty hack. Throughout the blockly code,
+  // constructors look at Blockly.readOnly to determine if they're
+  // supposed to be readonly. Because we want this BlockSpace to be
+  // readonly regardless of the global setting, we temporarily override
+  // the value while we build what we want, then restore it at the end.
+  // Similarly for languageTree, which is supposed to define the toolbox
+  var readOnly = Blockly.readOnly;
+  var languageTree = Blockly.languageTree;
+
+  Blockly.readOnly = true;
+  Blockly.languageTree = false;
+
+  this.blockSpaceEditor = new Blockly.BlockSpaceEditor(this.div, function () {
+    var metrics = Blockly.BlockSpaceEditor.prototype.getBlockSpaceMetrics_.call(this);
+    if (!metrics) {
+      return null;
+    }
+    // Expand the view so we don't see scrollbars
+    metrics.viewHeight += Blockly.BlockSpace.SCROLLABLE_MARGIN_BELOW_BOTTOM;
+    return metrics;
+  }, function (xyRatio) {
+    Blockly.BlockSpaceEditor.prototype.setBlockSpaceMetrics_.call(this, xyRatio);
+  }, true);
+
+  var blockSpace = this.blockSpaceEditor.blockSpace;
+  var parsedXml = parseXmlElement(this.xml);
+  Blockly.Xml.domToBlockSpace(blockSpace, parsedXml);
+
+  Blockly.readOnly = readOnly;
+  Blockly.languageTree = languageTree;
 };
 
-FeedbackBlocks.prototype.hideDiv = function() {
-  this.div.className += " hiddenIframe";
+FeedbackBlocks.prototype.show = function () {
+  this.div.style.visibility = '';
+  this.div.style.height = '';
```

Comment:
Would we be better off just doing $(this.div).show(), and elsewhere .hide()? Not sure of the answer